frontend/src/components/widgets/DataFrame/DataFrame.tsx

```diff
@@ -315,13 +316,86 @@ export interface DataFrameProps {
   data: Quiver
   width: number
   height?: number
+  isFullScreen: boolean
+}
+
+function getDefaultHeight(
+  element: ArrowProto,
+  containerHeight: number | undefined,
+  maxHeight: number,
+  isFullScreen: boolean
+): number {
+  let height = Math.min(maxHeight, DEFAULT_TABLE_HEIGHT)
+  if (element.height) {
+    // User has explicitly configured a height
+    height = Math.max(element.height, MIN_TABLE_HEIGHT)
+  }
+
+  if (containerHeight) {
+    // If container height is set (e.g. when used in fullscreen)
+    // The maxHeight and height should not be larger than container height
+    height = Math.min(height, containerHeight)
+
+    if (!element.height) {
+      // If no explicit height is set, set height to max height (fullscreen mode)
+      height = maxHeight
+    }
+  }
+
+  if (isFullScreen) {
+    height = height > maxHeight ? maxHeight : height
+  }
+  return height
+}
+
+function getMaxHeight(
+  element: ArrowProto,
+  containerHeight: number | undefined,
+  numRows: number
+): number {
+  // Automatic table height calculation: numRows +1 because of header, and +3 pixels for borders
+  let maxHeight = Math.max((numRows + 1) * ROW_HEIGHT + 3, MIN_TABLE_HEIGHT)
+  if (element.height) {
+    // User has explicitly configured a height
+    maxHeight = Math.max(element.height, maxHeight)
```

Comment:
I think this should be `Math.min`. If you set the element's height to be a large number, we will set the max height to be that number which I don't think we want.